src/analysis/state_helper.rs

```diff
@@ -69,6 +69,12 @@ pub fn comparison_segment_time(
     comparison: &str,
     method: TimingMethod,
 ) -> Option<TimeSpan> {
+    let segment_index = if segment_index >= run.len() {
+        run.len() - 1
+    } else {
+        segment_index
+    };
+
```

Comment:
This definitely shouldn't be here. The detailed timer only does this because we still show the segment you just finished at the end of the run. No other component does this though, so the segment time component shouldn't do this either. (We may want to change this, but that should be done for all components in a separate pull request then).

Reply:
The detailed timer does additional stuff in its equivalent function that was used before, such as choosing the best segment time instead of subtracting the split times for the best segments comparison. That should probably be done here too, such that the detailed timer can use this instead of its own copy. (cc @wooferzfg)

Reply:
That obviously changes the behavior in the splits component, but that's probably what we want anyway. That way it's consistent across the segment time component, detailer timer and the splits component.

Reply:
Yeah, it would be a lot better if these calculations were standardized across components.

However, I'm not sure I agree that this component shouldn't show the last segment's time after the run is over. The previous segment and delta components both show info about the last segment once the run ends. Possible time save doesn't show any info, but that's arguably not the behavior that we want. I would prefer to always show info about the last segment when the run ends rather than just showing nothing.

Reply:
I think we should still postpone that until it‘s clear which components could benefit from it. I‘d rather land this without the logic for now and we can decide what to do in subsequent PRs.

Reply:
@CryZe, I added this logic here because it prevents an index out of range panic if the caller were to pass a split index `>= run.len()`. This has no effect on the behavior of the splits component as its just clamping the split index to `run.len() - 1`.

Either way this logic needs to be *somewhere*, either here, or in the component itself otherwise LSC will panic after the run ends.

Reply:
It should be in the component. It shouldn‘t query the information for the current segment‘s time if there‘s no current segment. Handling wrong inputs by just clamping them is generally just causing hidden bugs.

Reply:
I moved the logic into the component and documented the panic.